src/java.desktop/windows/classes/sun/awt/windows/ThemeReader.java

```diff
@@ -111,11 +121,16 @@ private static Long getTheme(String widget) {
             try {
                 if (!valid) {
                     // Close old themes.
-                    for (Long value : widgetToTheme.values()) {
-                        closeTheme(value);
+                    if (!dpiAwareWidgetToTheme.isEmpty()) {
+                        for (Map<String, Long> dpiVal : dpiAwareWidgetToTheme.values()) {
+                            for (Long value : dpiVal.values()) {
+                                closeTheme(value);
+                            }
+                        }
+                        dpiAwareWidgetToTheme.get(dpi).clear();
+                        dpiAwareWidgetToTheme.clear();
+                        valid = true;
```

Comment:
The `valid` flag should be set to `true` even if the map was empty. Otherwise, it may never be set to `true`. One more point to dropping `if (!isEmpty())`.